### PR TITLE
FC-0001: Remove EdxRestApiClient usage in tubular

### PR DESCRIPTION
**Pull request by ruzniaievdm**
JIRA: [DEPR-138](https://openedx.atlassian.net/browse/DEPR-138)
https://github.com/openedx/public-engineering/issues/38

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ruzniaievdm! I've created [OSPR-6628](https://openedx.atlassian.net/browse/OSPR-6628) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Review by nadeemshahzad — APPROVED** [review]

**Comment by dianakhuang** [comment]
You need to remove the runs of 3.6 and 3.7 from https://github.com/openedx/tubular/blob/master/.github/workflows/tox.yml#L10 as well.

**Comment by openedx-webhooks** [comment]
@ruzniaievdm 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tubular&entry.867055334=https://github.com/openedx/tubular/pull/558&entry.1484655318=https://github.com/ruzniaievdm&entry.752974735=2022-04-19+09:34&entry.1917517419=2022-04-25+15:55&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by HassanJaveed84** [comment]
@dianakhuang @nadeemshahzad this seems to be breaking our retirement pipeline. 

http://jenkins-new.analytics.edx.org/job/user-retirement-driver/11418/console

```
12:09:14-0500 + python scripts/retire_one_learner.py --config_file=/var/lib/jenkins/workspace/user-retirement-driver/user-retirement-secure/prod-edx.yml
12:09:15-0500 b'Learner Retirement: Starting learner retirement for <USERNAME> using config file /var/lib/jenkins/workspace/user-retirement-driver/user-retirement-secure/prod-edx.yml'
12:09:16-0500 ERROR:tubular.edx_api:API Error: 502 Server Error: Bad Gateway for url: https://ecommerce.edx.org/api/v2/retirement/tracking_id<USERNAME>/ with status code: 502

**Comment by ruzniaievdm** [comment]
It’s my bad it looks like you getting an error due to trailing slash before `<USERNAME>/`  in url `https://ecommerce.edx.org/api/v2/retirement/tracking_id<USERNAME>/`

**Comment by HassanJaveed84** [comment]
Reverting this for now to get the pipeline running.

File: tubular/edx_api.py [hunk 1]
```diff
@@ -370,18 +392,17 @@ def retire_learner(self, learner):
...
     @_retry_lms_api()
     def get_tracking_key(self, learner):
         """
         Fetches the ecommerce tracking id used for Segment tracking when
         ecommerce doesn't have access to the LMS user id.
         """
-        with correct_exception():
-            result = self._client.api.v2.retirement.tracking_id(learner['original_username']).get()
-            return result['ecommerce_tracking_id']
+        api_url = self.get_api_url(f"api/v2/retirement/tracking_id{learner['original_username']}")
```
Comment (dianakhuang) [inline tubular/edx_api.py:404]: This needs to be `f"api/v2/retirement/tracking_id/{learner['original_username']}"`

File: tubular/scripts/retire_one_learner.py [hunk 2]
```diff
@@ -22,29 +23,23 @@
...
-import sys
 
 import click
-from six import text_type
-from slumber.exceptions import HttpNotFoundError
 
 # Add top-level module path to sys.path before importing tubular code.
+from tubular.exception import HttpDoesNotExistException
+
 sys.path.append(path.dirname(path.dirname(path.abspath(__file__))))
```
Comment (HassanJaveed84) [inline tubular/scripts/retire_one_learner.py:37]: The import needs to come after `sys.path.append`

Comment (ruzniaievdm) [reply tubular/scripts/retire_one_learner.py:37]: done

**Comment by ruzniaievdm** [comment]
reopened PR - https://github.com/openedx/tubular/pull/562
In accordance with all discussions

**Comment by natabene** [comment]
@ruzniaievdm @dianakhuang @nadeemshahzad  Thank you all!

**Pull request merged**